src/Controls/src/Xaml/ApplyPropertiesVisitor.cs

```diff
@@ -331,21 +331,16 @@ static string GetContentPropertyName(IEnumerable<CustomAttributeData> attributes
 		{
 			// F# does not support public fields, so allow internal (Assembly) as well as public
 			const BindingFlags supportedFlags = BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.FlattenHierarchy;
-			var bindableFieldInfo = elementType.GetFields(supportedFlags)
-												.FirstOrDefault(fi => (fi.IsAssembly || fi.IsPublic) && fi.Name == localName + "Property");
-
-			Exception exception = null;
-			if (exception == null && bindableFieldInfo == null)
+			var bindableFieldInfo = elementType.GetField(localName + "Property", supportedFlags);
+			if (bindableFieldInfo is not null && (bindableFieldInfo.IsAssembly || bindableFieldInfo.IsPublic))
 			{
-				exception =
-					new XamlParseException(
-						Format("BindableProperty {0} not found on {1}", localName + "Property", elementType.Name), lineInfo);
-			}
-
-			if (exception == null)
 				return bindableFieldInfo.GetValue(null) as BindableProperty;
+			}
 			if (throwOnError)
-				throw exception;
+			{
+				throw new XamlParseException(
```

Comment:
it even looks like all invocations of GetBindableProperty use false for throwOnError, so this could be simplified, and the signature changed to return a BP? so the intention are clear

Reply:
there's probably a story behind that, but even I can't remember it

Reply:
I removed the `throwOnError` parameter. 👍